Cherry-picks for 0.6.2 release

XVilka opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Bugfixes and documentation updates

**Test plan**

CI is green

Commit:
Fix memleak in rz_reg_item_free()

Commit:
Fix memleak in parse_reg_profile_str()

Commit:
Fix memleak in parse_relocs_from_indirect_syms()

Commit:
Fix memleak in rz_assembly_of_hex_handler()

PeiweiHu committed:
Update the clang-format version in DEVELOPERS.md (#3776)

XVilka committed:
Update capstone-v5 to 5.0.1 (#3777)

Commit:
Fix several memleaks (#3779)

* Fix memleak in rz_il_handler_empty()

* Fix memleak in compare_op()

* Fix memleak in assign_reg()

* Fix memleak in rz_asm_op_fini()

* Fix memleak in rz_core_analysis_bytes()

* Fix memleak in add_sub_op()

* Fix memleak in bitwise_op()

* Fix memleak in shift_and_rotate()

ur4t committed:
subprojects/zlib: Improve meson.build, emulating configure script (#3759)

* subprojects/zlib: Improve meson.build, emulating configure script
* Update zlib to 1.3

tinez committed:
Fix syscall preprocessing

'rizin' is missing a lot of syscalls for i.e. arm. We can check that
with:

 rizin -a x86_64 -b 64 -k linux -q -c asl

vs

 rizin -a arm -b 64 -k linux -q -c asl

'rz_syscall_item_new_from_string' splits the incoming string by commas
and expects at least 3 items as a result. This patch fixes
syscall preprocessing by appending empty strings to match the expected
number of items after a split.

This is a clone of a fix posted to radare2 [1].

[1] https://github.com/radareorg/radare2/pull/21508

tinez committed:
Remove duplicate entries from 'linux-arm-64.sdb.txt'

This is a cherry pick of [1]. The removed mappings overwrite previous
entries and cannot be found in [2].

[1] https://github.com/radareorg/radare2/commit/c67c47ed733437937e602626d1446c75539ee9ca
[2] https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/unistd.h

Signed-off-by: Marcin Sobczyk <msobczyk@redhat.com>

byteninjaa0 committed:
Actualize `rizin` and `rz-diff` man pages

laomaiweng committed:
Refactor the pointer command to use the API (#3788)

* Add test for pointer read command `*`

Test for `asm.bits=8` uses `asm.arch=6502` because `asm.bits=8` is
unsupported for the default arch.

* Fix pointer command `*` broken since move from `?` to `%` ##shell

* Refactor to use appropriate existing API in `num_callback`

kazarmy committed:
Fix cmd_pointer test on Travis (#3791)

XVilka commented:
Looks like only missing things are remaining `?v` invocations @ret2libc

p371jah commented:
Maybe these changes too #3752

XVilka commented:
@pelijah I already imported them.

XVilka commented:
@pelijah sorry, I squashed them, didn't notice the title

p371jah commented:
No it is older one fix pack. You imported #3779 which is newer. I don't see commits from #3752 in stable.

ttxine committed:
Fix segfault in rz_seek_end_handler (#3797)

laomaiweng committed:
Restore original seek offset after iterating over flags with `@@f` ##shell (#3794)

* Update the golang analysis tests to seek to the main function before
  dumping strings, which was previously done as a side effect of using
  `@@f` during auto-analysis.

ret2libc commented on librz/include/rz_util/rz_strbuf.h:
```diff
@@ -12,7 +12,6 @@ typedef struct {
 	size_t len; // string length in chars or binary buffer size
 	char *ptr; // ptr replacing buf in case strlen > sizeof(buf)
 	size_t ptrlen; // string length + 1 or binary buffer size
-	bool weakref; // ptr is not owned
```
@XVilka this can't be done in a patch release, in theory. It should be done in the next release.

XVilka replied:
Good point, will remove that commit

kazarmy committed:
rz-run resolve_value: Pass single-quoted strings unaltered (#3800)

* Test whether there is error on `rizin -d`

* Make env more like local env

* Add space btw `env` and its arg

* rz-run resolve_value: Pass single-quoted strings unaltered

ttxine committed:
Fix div by zero in 's++' and 's--' (#3803)

ttxine committed:
Fix several memleaks (#3811)

* Fix memleak in rz_seek_history_list_handler()
* Fix memleaks in rz_eval_getset_handler()
* Fix memleaks in visual_comma()
* Fix memleak in cmd_pxr()
* Fix memleak in arged_stmt

byteninjaa0 committed:
Fix rz-diff help message (#3814)

PeiweiHu committed:
Fix:  wrong exit condition in rz_analysis_optype_from_string (#3821)

devnexen committed:
bsd_generate_corefile for FreeBSD adjust version since the feature is available in the 13.2 serie (#3820)

ifyGecko committed:
fixed pyc uaf (#3823)

XVilka commented:
Something is really wrong with NetBSD (happens also on `dev`):
```
+ python3.8 -m ensurepip --user Traceback (most recent call last):   File "<string>", line 6, in <module>   File "/usr/pkg/lib/python3.8/runpy.py", line 207, in run_module     return _run_module_code(code, init_globals, run_name, mod_spec)   File "/usr/pkg/lib/python3.8/runpy.py", line 97, in _run_module_code     _run_code(code, mod_globals, init_globals,   File "/usr/pkg/lib/python3.8/runpy.py", line 87, in _run_code     exec(code, run_globals)   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/__main__.py", line 31, in <module>   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/_internal/cli/main.py", line 68, in main   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/_internal/commands/__init__.py", line 114, in create_command   File "/usr/pkg/lib/python3.8/importlib/__init__.py", line 127, in import_module     return _bootstrap._gcd_import(name[level:], package, level)   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import   File "<frozen importlib._bootstrap>", line 991, in _find_and_load   File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked   File "<frozen importlib._bootstrap>", line 655, in _load_unlocked   File "<frozen importlib._bootstrap>", line 618, in _load_backward_compatible   File "<frozen zipimport>", line 259, in load_module   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/_internal/commands/install.py", line 16, in <module>   File "<frozen zipimport>", line 259, in load_module   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/_internal/cli/req_command.py", line 21, in <module>   File "<frozen zipimport>", line 259, in load_module   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/_internal/index/package_finder.py", line 30, in <module>   File "<frozen zipimport>", line 259, in load_module   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/_internal/req/__init__.py", line 8, in <module>   File "<frozen zipimport>", line 259, in load_module   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/_internal/req/req_install.py", line 44, in <module>   File "<frozen zipimport>", line 259, in load_module   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/_internal/operations/install/wheel.py", line 39, in <module>   File "<frozen zipimport>", line 259, in load_module   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/_vendor/distlib/scripts.py", line 16, in <module>   File "<frozen zipimport>", line 259, in load_module   File "/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl/pip/_vendor/distlib/compat.py", line 83, in <module>   File "/usr/pkg/lib/python3.8/xmlrpc/client.py", line 138, in <module>     from xml.parsers import expat   File "/usr/pkg/lib/python3.8/xml/parsers/expat.py", line 4, in <module>     from pyexpat import * ModuleNotFoundError: No module named 'pyexpat' Traceback (most recent call last):   File "/usr/pkg/lib/python3.8/runpy.py", line 194, in _run_module_as_main     return _run_code(code, main_globals, None,   File "/usr/pkg/lib/python3.8/runpy.py", line 87, in _run_code     exec(code, run_globals)   File "/usr/pkg/lib/python3.8/ensurepip/__main__.py", line 5, in <module>     sys.exit(ensurepip._main())   File "/usr/pkg/lib/python3.8/ensurepip/__init__.py", line 209, in _main     return _bootstrap(   File "/usr/pkg/lib/python3.8/ensurepip/__init__.py", line 128, in _bootstrap     return _run_pip(args + [p[0] for p in _PROJECTS], additional_paths)   File "/usr/pkg/lib/python3.8/ensurepip/__init__.py", line 37, in _run_pip     return subprocess.run(cmd, check=True).returncode   File "/usr/pkg/lib/python3.8/subprocess.py", line 516, in run     raise CalledProcessError(retcode, process.args, subprocess.CalledProcessError: Command '['/usr/pkg/bin/python3.8', '-c', '\nimport runpy\nimport sys\nsys.path = [\'/tmp/tmpcfwb944t/setuptools-56.0.0-py3-none-any.whl\', \'/tmp/tmpcfwb944t/pip-23.0.1-py3-none-any.whl\'] + sys.path\nsys.argv[1:] = [\'install\', \'--no-cache-dir\', \'--no-index\', \'--find-links\', \'/tmp/tmpcfwb944t\', \'--user\', \'setuptools\', \'pip\']\nrunpy.run_module("pip", run_name="__main__", alter_sys=True)\n']' returned non-zero exit status 1.
```
The crux of the problem seems to be missing `pyexpat` module for some reason: `ModuleNotFoundError: No module named 'pyexpat'`

XVilka committed:
Update rz-libdemangle (#3827)

kazarmy committed:
Add `py38-expat` package to netbsd.yml (#3836)
